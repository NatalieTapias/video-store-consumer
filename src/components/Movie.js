import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Movie extends Component {

  constructor() {
    super();

    this.state = {
      showDetails: false,
    }
  };

  onClickToggleShowDetails = () => {
    let doesItShowDetails = this.state.showDetails;
    doesItShowDetails = !(this.state.showDetails);
    this.setState({showDetails: doesItShowDetails});
  }

  render() {
    const {title, image_url, overview, release_date, selectMovieCallback, id} = this.props;

    return (
      <a onClick={this.onClickToggleShowDetails} className="col-sm-2">
          <div className="card">
            <h3 className="card-title">{title}</h3>
            <div className="card-body">
              <img src={image_url} alt={title} /> 
              {this.state.showDetails && ( 
                <>
                <p className="card-text">{overview}</p>
                <p>Released on: {release_date}</p>
                </>
              )}
              <button
                className="btn btn-primary"
                onClick={() => { selectMovieCallback(id) }}>
                Select
              </button>
            </div>
          </div>  
      </a>
      );
    }
  };

export default Movie;

Movie.propTypes = {
  title: PropTypes.string,
  image_url: PropTypes.string, 
  overview: PropTypes.string, 
  release_date: PropTypes.string, 
  selectMovieCallback: PropTypes.func, 
  id: PropTypes.number,
};