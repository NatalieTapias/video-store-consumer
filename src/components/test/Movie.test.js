import React from 'react';
import { render, cleanup } from '@testing-library/react'
import Movie from '../Movie';

describe('Movie', () => {
  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <Movie 
      id={3}
      key={3}
      title={"excellent movie"}
      overview={"movie.overview"}
      release_date={"movie.release_date"}
      image_url={"movie.image_url"}
      external_id={"movie.external_id"}
      selectMovieCallback={() => {}} 
      /> 
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
});