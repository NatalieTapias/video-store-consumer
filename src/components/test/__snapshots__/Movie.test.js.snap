// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <a
    class="col-sm-2"
  >
    <div
      class="card card-expanded"
    >
      <h3
        class="card-title"
      >
        excellent movie
      </h3>
      <div
        class="card-body"
      >
        <img
          alt="excellent movie"
          src="movie.image_url"
        />
        <div>
          <button
            class="btn btn-primary"
          >
            Select
          </button>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
